plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -347,10 +347,10 @@ else if (type instanceof ObjectType) {
                 .map(index -> new SchemaTableName(this.schemaName, index))
                 .forEach(result::add);
 
-        client.getAliases().entrySet().stream()
-                .filter(entry -> indexes.contains(entry.getKey()))
-                .flatMap(entry -> entry.getValue().stream()
-                        .map(alias -> new SchemaTableName(this.schemaName, alias)))
+        client.getAliases().entrySet()
```

Comment:
`getAliases` doesn't need to change, and all you need is to restructure this to dedup the names before constructing the list. You can do it like this:

```java
        Set<SchemaTableName> result = new HashSet<>();
        
        indexes.stream()
                ...
                .forEach(result::add);

        client.getAliases().entrySet().stream()
                ...
                .forEach(result::add);

        return ImmutableList.copyOf(result);
```

Reply:
Currently `getAliases()` computes and returns Map(Index, List(Alias)). So whoever fetches aliases in the future also, will have to dedup every time. But with this PR, we compute and return Map(Alias, List(Indices)), which makes more sense. No need to dedup. No need to iterate through the entire Map to identify all the indices of an alias.

Is there any specific significance why we return Map(Index, List(Alias)) currently?

Reply:
I think @martint wants to keep these calls aligned with the structure that Elasticsearch returns to make the client simple to understand for others. `getAliases()` and other calls in the client are in essence dumb wrappers that just maps the rest call response to in-mem Java structure. From there it's up to us to map it however we like. If this is going to add a lot of extra work, then we can cache the another structure in the client or somewhere else if necessary. It doesn't look like a lot to me though.

Reply:
Got it. Only dedup is handled now.